src/main/webapp/cdn/blockly/generators/propc/control.js

```diff
@@ -106,10 +106,10 @@ Blockly.propc.control_repeat_for_loop = function () {
     var loop_counter = Blockly.propc.variableDB_.getName(this.getFieldValue('VAR'), Blockly.Variables.NAME_TYPE);
 
     if (Number(start) < Number(end)) {
-        var code = 'for (' + loop_counter + ' = ' + start + '; ' + loop_counter + ' < ' + end + '; ' + loop_counter + ' += ' + step + ') {\n' + repeat_code + '\n}';
+        var code = 'for (' + loop_counter + ' = ' + start + '; ' + loop_counter + ' <= ' + end + '; ' + loop_counter + ' += ' + step + ') {\n' + repeat_code + '\n}';
         return code;
     } else {
-      var code = 'for (' + loop_counter + ' = ' + start + '; ' + loop_counter + ' > ' + end + '; ' + loop_counter + ' += ' + step + ') {\n' + repeat_code + '\n}';
+      var code = 'for (' + loop_counter + ' = ' + start + '; ' + loop_counter + ' >= ' + end + '; ' + loop_counter + ' += ' + step + ') {\n' + repeat_code + '\n}';
```

Comment:
Ln 112 - If start => end, then the condition __loop_counter >= end__ will _always_ be true when the modifier __loop_counter += step__ has a positive _step_ value.